src/generator.cs

```diff
@@ -1382,6 +1382,10 @@ string GetToBindAsWrapper (MemberInformation minfo = null, ParameterInfo pi = nu
 			temp = isNullable ? $"{parameterName} == null ? null : " : string.Empty;
 			temp += $"{FormatType (retType.DeclaringType, retType)}Extensions.GetConstant ({parameterName}{denullify});";
 		} else if (originalType.IsArray && originalType.GetArrayRank () == 1) {
+			if (!retType.IsArray) {
+				throw new BindingException (1072, true, "The BindAs type for the parameter \"{0}\" in the method \"{1}.{2}\" must be an array when the parameter's type is an array.",
+					parameterName,mi.ReturnType.Name, mi.Name);
```

Comment:
The type name is wrong :smile: I think this is right:

```suggestion
					parameterName, mi.DeclaringType.FullName, mi.Name);
```

Reply:
Oops, misunderstood your previous comment. I think it's correct now.